core/src/test/scala/org/apache/spark/rdd/RDDBarrierSuite.scala

```diff
@@ -29,6 +29,15 @@ class RDDBarrierSuite extends SparkFunSuite with SharedSparkContext {
     assert(rdd2.isBarrier())
   }
 
+  test("RDDBarrier mapPartitionsWithIndex") {
+    val rdd = sc.parallelize(1 to 12, 4)
+    assert(rdd.isBarrier() === false)
+
+    val rdd2 = rdd.barrier().mapPartitionsWithIndex((index, iter) => Iterator(index, iter.sum))
+    assert(rdd2.isBarrier())
+    assert(rdd2.collect().toList === List((0, 6), (1, 15), (2, 24), (3, 33)))
```

Comment:
This might be true for now, but the result might change in the future, how about just check the index of the result?

Reply:
thanks for reviewing, updated it.